#! /bin/sh /usr/share/dpatch/dpatch-run
## 04optparser.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix for #595567, option parser errors out on some options 

@DPATCH@
diff -urNad duplicity-0.6.09~/src/commandline.py duplicity-0.6.09/src/commandline.py
--- duplicity-0.6.09~/src/commandline.py	2010-07-26 02:14:39.000000000 +1000
+++ duplicity-0.6.09/src/commandline.py	2010-09-06 12:55:31.969750215 +1000
@@ -312,6 +312,7 @@
     # circumstances. the default should absolutely always be False unless
     # you know what you are doing.
     parser.add_option("--ignore-errors", action="callback",
+                      dest="ignore_errors",
                       callback=lambda o, s, v, p: (log.Warn(
                           _("Running in 'ignore errors' mode due to %s; please "
                             "re-consider if this was not intended") % s),
@@ -368,6 +369,7 @@
 
     # Whether the old filename format is in effect.
     parser.add_option("--old-filenames", action="callback",
+                      dest="old_filenames",
                       callback=lambda o, s, v, p: (setattr(p.values, o.dest, True),
                                                    old_fn_deprecation(s)))
 
@@ -403,6 +405,7 @@
 
     # If set, use short (< 30 char) filenames for all the remote files.
     parser.add_option("--short-filenames", action="callback",
+                      dest="short_filenames",
                       callback=lambda o, s, v, p: (setattr(p.values, o.dest, True),
                                                    old_fn_deprecation(s)))
 
